resources/mixins/kubernetes/mixin.libsonnet

```diff
@@ -9,6 +9,7 @@ kubernetes {
     kubeApiserverSelector: 'job="api"',
     kubeProxySelector: 'job="machine-config-daemon"',
     kubeSchedulerSelector: 'job="scheduler"',
+    namespaceSelector: 'namespace=~"rhacs-.*|acscs-.*"',
```

Comment:
I also found the following possibly interesting and one definitely interesting namespace in `acs-stage-dp-02` that don't match your regex:
- `dinosaur-counter`
- `external-secrets`
-  `rhacs`

Reply:
How can we confirm which namespaces we want to monitor (except the obvious `rhacs`)?

```
% kubectl get ns | grep -Ev '^(openshift|kube|rhacs-|acscs-)' 
NAME                                               STATUS   AGE
dedicated-admin                                    Active   216d
default                                            Active   216d
dinosaur-counter                                   Active   195d
external-secrets                                   Active   19d
qa                                                 Active   125d
rhacs                                              Active   202d
```

Reply:
Should we blocklist instead, e.g.: `namespace!~"openshift-.*|kube-.*"`?

Reply:
I'm 100% sure we want `rhacs`. @kovayur would know about `external-secrets`.
I don't have any good ideas about the rest other than asking on slack.

However this highlights an important point - how do we make sure going forward that this list is up to date as the system evolves? Maybe a denylist is indeed a more robust solution. Better to have to add to it than miss something important :thinking:

Reply:
I think it makes sense to ignore `openshift-.*` and `kube-.*` namespaces. It's highly unlikely to use those for new deployment so no need to think about keeping the list up to date

Reply:
I think even `openshift-.*` might be too broad. For example, we use OpenShift logging operator for ACS-CS logging, which is installed in `openshift-logging`. Another candidate would be `openshift-authentication`, which we use for authn/z to access the cluster. I would suggest to start with a conservative deny list that takes care of the alerts that annoyed us so far, and expand it as we go.

Reply:
@porridge:

> how do we make sure going forward that this list is up to date as the system evolves?

If we have control over the name of the namespaces we will need, I would say an allowlist has the benefit of the namespace creator to decide if he wants to have his workload monitored or not (by prefixing `rhacs-` or `acscs-`).

Reply:
@stehessel:

> I think even openshift-.* might be too broad. For example, we use OpenShift logging operator for ACS-CS logging, which is installed in openshift-logging

The idea behind blocking `openshift-` is that SRE already has monitoring that covers platform workloads.   That being said, SRE does not monitor every pod.  They have symptom-based alerts that *should* cover issues with platform components that will direct to the specific causes.

> I would suggest to start with a conservative deny list that takes care of the alerts that annoyed us so far, and expand it as we go.

Maybe we should think if we currently have any actionable steps for the `openshift-` alarms when they are triggered and not impacting ACSCS. Do you think it would be more practical to use alarms that monitor the impact in ACSCS instead?

I agree that some of the `openshift-` alarms might be interesting to monitor, but these would be the exceptions, AFAIK. We have ~100 openshfit namespaces. Also, I think the static pods that annoyed us might exist in multiple namespaces. In the latest alarm they were all `openshift-kube-.*`, though. So maybe block that instead for now, with `kube-.*`.

Reply:
Moving to blocklist as suggested by me above, we have the following list in `eu-01`:

```
% kubectl get ns | grep -E '^(openshift-kube|kube)' | awk '{print $1}'
kube-node-lease
kube-public
kube-system
openshift-kube-apiserver
openshift-kube-apiserver-operator
openshift-kube-controller-manager
openshift-kube-controller-manager-operator
openshift-kube-scheduler
openshift-kube-scheduler-operator
openshift-kube-storage-version-migrator
openshift-kube-storage-version-migrator-operator
%
```